web-searchers/langchain4j-web-search-google-customsearch/src/test/java/dev/langchain4j/web/search/google/customsearch/GoogleCustomWebSearchEngineIT.java

```diff
@@ -0,0 +1,136 @@
+package dev.langchain4j.web.search.google.customsearch;
+
+import dev.langchain4j.web.search.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@EnabledIfEnvironmentVariable(named = "GOOGLE_API_KEY", matches = ".*")
+@EnabledIfEnvironmentVariable(named = "GOOGLE_SEARCH_ENGINE_ID", matches = ".*")
+class GoogleCustomWebSearchEngineIT extends WebSearchIT {
+
+    WebSearchEngine googleSearchEngine = GoogleCustomWebSearchEngine.withApiKeyAndCsi(
+                                            System.getenv("GOOGLE_API_KEY"),
+                                            System.getenv("GOOGLE_SEARCH_ENGINE_ID"));
+
+    @Test
+    void should_return_default_web_results_with_search_information() {
+        // given
+        String query = "What is the weather in New York?";
+
+        // when
+        WebSearchResults results = googleSearchEngine.search(query);
+
+        // then
+        assertThat(results.searchMetadata()).isNotNull();
+        assertThat(results.searchInformation().totalResults()).isGreaterThan(0);
+        assertThat(results.results().size()).isGreaterThan(0);
+    }
+
+    @Test
+    void should_return_safe_web_results_in_spanish_language() {
+        // given
+        String query = "Who won the FIFA World Cup 2022?";
+        WebSearchRequest webSearchRequest = WebSearchRequest.builder()
+                .searchTerms(query)
```

Comment:
WDYT about renaming `searchTerms` into `query` in `WebSearchRequest`? We use `query` pretty much everywhere

Reply:
uhm yeah could be. I'll add some little changes in `WebSearchRequest` and `WebSearchOrganicResult` to pass images related to the use query to include it as part of `TextSegment` and `Tool` as well. I'll include it in other PR.  :)